src/Illuminate/Support/Arr.php

```diff
@@ -256,11 +256,11 @@ public static function forget(&$array, $keys)
             while (count($parts) > 1) {
                 $part = array_shift($parts);
 
-                if (isset($array[$part]) && is_array($array[$part])) {
-                    $array = &$array[$part];
-                } else {
+                if (! isset($array[$part]) || ! is_array($array[$part])) {
```

Comment:
This is harder to read `if true and true` is in my humble opinion better than `if not false or not false`